WordPress/src/main/java/org/wordpress/android/ui/sitecreation/theme/DesignPreviewFragment.kt

```diff
@@ -84,15 +105,26 @@ class DesignPreviewFragment : FullscreenBottomSheetDialogFragment() {
             }
         })
 
+        viewModel.previewMode.observe(viewLifecycleOwner, Observer { load() })
+
+        viewModel.onPreviewModeButtonPressed.observe(viewLifecycleOwner, Observer {
+            previewModeSelectorPopup.show(viewModel)
+        })
+
+        previewModeSelectorPopup = PreviewModeSelectorPopup(requireActivity(), previewTypeSelectorButton)
+
         backButton.setOnClickListener { closeModal() }
 
         chooseButton.setOnClickListener { viewModel.onPreviewChooseTapped() }
 
+        previewTypeSelectorButton.setOnClickListener { viewModel.onPreviewModePressed() }
+
         webView.settings.userAgentString = WordPress.getUserAgent()
         webView.webViewClient = object : WebViewClient() {
             override fun onPageFinished(view: WebView?, url: String?) {
                 super.onPageFinished(view, url)
-                viewModel.onPreviewLoaded(template)
+                val widthScript = getString(R.string.web_preview_width_script, viewModel.getPreviewMode().previewWidth)
+                view?.evaluateJavascript(widthScript) { viewModel.onPreviewLoaded(template) }
```

Comment:
It seems this may need a guard when the view is destroyed before the `WebView` finishes loading, since there is no context when the fragment is detached.

#### Steps to reproduce crash:
* Tap a design
* Tap preview
* Quickly tap the back button (before the preview finishes loading)
* :boom: 

These changes resolved it for me:

```suggestion
                val widthScript = context?.getString(R.string.web_preview_width_script, viewModel.getPreviewMode().previewWidth)
                if (widthScript != null) {
                    view?.evaluateJavascript(widthScript) { viewModel.onPreviewLoaded(template) }
                }
```

Reply:
Nice catch @mkevins 👍 
I tested the suggested solution and resolves the problem 🙇 
I did a minor edit to handle `max line length` lint issue.